src/workers/parser/getOutOfScopeLocations.js

```diff
@@ -83,13 +83,12 @@ function getOutOfScopeLocations(
 ): AstLocation[] {
   const { functions, comments } = findSymbols(source);
   const commentLocations = comments.map(c => c.location);
-
-  return functions
+  const locations = functions
     .map(getLocation)
     .concat(commentLocations)
-    .filter(loc => !containsPosition(loc, position))
-    .reduce(removeOverlaps, [])
-    .sort(sortByStart);
+    .sort(sortByStart)
+    .filter(loc => !containsPosition(loc, position));
+  return removeOverlaps(locations);
```

Comment:
`sort` should be O(nlogn). The old `removeOverlaps` is O(n^2) so the old way was doing O(n^2) then O(nlogn).
Try to improve that, just do once O(nlogn) then the following operations can do O(n)